### ISSUE TITLE
[firebase_analytics] Version 5+ causes app to immediately close

### ISSUE THREAD

**Issue by timeshift-ted** [FirebaseExtended/flutterfire#1595]
I have installed this plug-in and notice that if I use version 5 or higher, the app immediately closes as soon as it opens (splash screen appears for about one second and then it closes). No error is logged, and this seems to happen before a debugging connection can be made by Android Studio.

After seeing a YouTube video by someone with the same issue (https://www.youtube.com/watch?v=SXuB9oa3t9A) I decided to just try an older version, even though I've never used this plug-in before.

I am only building this for Android, and testing is being done on a physical lPixel 1 phone running Android 10.

I found that versions lower than 5 work OK, however they give these warnings on build:

```bash
Note: /Users/Ted/Documents/Timeshift/Development/flutter/.pub-cache/hosted/pub.dartlang.org/firebase_analytics-4.0.2/android/src/main/java/io/flutter/plugins/firebaseanalytics/FirebaseAnalyticsPlugin.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: /Users/Ted/Documents/Timeshift/Development/flutter/.pub-cache/hosted/pub.dartlang.org/firebase_analytics-4.0.2/android/src/main/java/io/flutter/plugins/firebaseanalytics/FirebaseAnalyticsPlugin.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
```

I am up to date on flutter upgrade, and flutter doctor doesn't show any issues.

```bash
[✓] Flutter (Channel stable, v1.9.1+hotfix.6, on Mac OS X 10.15.1 19B88, locale
    en-US)
 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
[✓] Xcode - develop for iOS and macOS (Xcode 11.1)
[✓] Android Studio (version 3.5)
[✓] Connected device (1 available)

• No issues found!
```

Would love to use the current version... if anyone has any suggestions I would very much appreciate it.

**Comment by iapicca** [comment]
Hi @phoelapyae69 
can you please provide your `flutter run --verbose`?

I'm not suggesting it or recommending this, 
but just for science, you might want to try switching channel
`
 flutter channel master
`
`
 flutter upgrade
`
Thank you

**Comment by timeshift-ted** [comment]
When I ran `flutter upgrade` I noticed this message:
`Your flutter checkout has local changes that would be erased by upgrading. If you want to keep these changes, it is recommended that you stash them via "git stash" or else commit the changes to a local branch. If it is okay to remove local changes, then re-run this command with --force.`
So I had a look in git and there were some changed files... So my guess is my previous attempt to upgrade didn't complete properly. 

I first tried the --force option but still got the same error.

Next, I committed those changed files to a local branch and then did another `flutter upgrade` and I noticed that took me from v1.9.1+hotfix.6 to 1.12.13+hotfix.5 and seemed to work OK.

Now, the analytics plugin seems to be working (still on stable branch).

Still don't know what caused those file changes (I don't mess with those files).

Just to be clear, am I meant to have firebase_core AND firebase_analytics in my pubspec.yaml? These's some documentation that shows both, and other that shows just firebase_analytics.
